consensus/safety-rules/src/safety_rules_2chain.rs

```diff
@@ -90,6 +93,29 @@ impl SafetyRules {
         Ok(vote)
     }
 
+    pub(crate) fn guarded_construct_and_sign_order_vote(
+        &mut self,
+        order_vote_proposal: &OrderVoteProposal,
+    ) -> Result<OrderVote, Error> {
+        // Exit early if we cannot sign
+        self.signer()?;
+        self.verify_order_vote_proposal(order_vote_proposal)?;
+        let proposed_block = order_vote_proposal.block();
+        let mut safety_data = self.persistent_storage.safety_data()?;
+
+        // Record 1-chain data
+        self.observe_qc(order_vote_proposal.quorum_cert(), &mut safety_data);
```

Comment:
I don't see we persist the safety data back to storage like `self.persistent_storage.set_safety_data(safety_data)?;`, what's the missing test case that would cover it? this is also a safety violation

Reply:
`observe_qc` updates `one_chain_round` and `preferred_round` in safety data. These don't seem to be used in any safety rules. We are only updating them as we do voting. And order voting safety rules don't use these fields. We could actually remove the `observe_qc` call from here? cc @danielxiangzl

Reply:
No, we need to update the `one_chain_round` which is the highest QC round that the timeout message should contain. function `safe_to_timeout` uses `one_chain_round`, no?

Reply:
In suite.rs, I've added assertions to make sure after we observe a QC, we can't timeout on its previous round.